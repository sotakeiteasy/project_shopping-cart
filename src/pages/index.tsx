import Home from "./home/home";
import Catalog from "./catalog/catalog";
import Cart from "./cart/cart";

export { Home, Catalog, Cart };
