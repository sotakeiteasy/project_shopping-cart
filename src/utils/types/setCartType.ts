import { Product } from "./product";
export type SetCartType = React.Dispatch<React.SetStateAction<Product[]>>;
