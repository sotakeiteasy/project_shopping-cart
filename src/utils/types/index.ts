import { Product } from "./product";
import { SetCartType } from "./setCartType";

export type { Product, SetCartType };
