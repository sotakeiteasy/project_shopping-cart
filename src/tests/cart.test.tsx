import { render, screen } from "@testing-library/react";
import { Cart } from "@/pages";
import { Product } from "@/utils/types";

vi.mock('@/utils/hooks/useCart', () => ({
  useCart: () => ({
    cart: mockedCart,
  }),
}))

let mockedCart: Product[] = [];

describe("render cart", () => {
    afterEach(() => {
        vi.resetAllMocks()
        
    })
    
    it('snapshot cart', () => {
        const { container } = render(<Cart />)
        expect(container).toMatchSnapshot()
    })

    it('shows "Cart is empty" if no products', () => {
        mockedCart = []
        render(<Cart/>)
        expect(screen.getByText('Cart is empty')).toBeInTheDocument();
    })

    it('renders product if one returned', () => {
        mockedCart = [
            {
                id: 1,
                title: 'test',
                price: 20,
                description: "test",
                category: "test",
                image: "test",
                rating: { count: 2, rate: 5 },
            }
        ]
        render(<Cart/>)
        expect(screen.queryByText('Cart is empty')).not.toBeInTheDocument();
    })

    it('renders delivery price if total < 50$', () => {
        mockedCart = [
            {
                id: 1,
                title: 'test',
                price: 20,
                description: "test",
                category: "test",
                image: "test",
                rating: { count: 2, rate: 5 },
            }
        ]
        render(<Cart/>)
        expect(screen.getByText('Delivery:')).toBeInTheDocument();
        expect(screen.getByText('5$')).toBeInTheDocument();
    })

    it('renders delivery price if total < 50$', () => {
        mockedCart = [
            {
                id: 1,
                title: 'test',
                price: 51,
                description: "test",
                category: "test",
                image: "test",
                rating: { count: 2, rate: 5 },
            }
        ]
        render(<Cart/>)
        expect(screen.getByText('Delivery:')).toBeInTheDocument();
        expect(screen.getByText('0$')).toBeInTheDocument();
    })

        it('renders disabled button if cart is empty', () => {
        mockedCart = []
        render(<Cart/>)
        expect(screen.getByRole('button', {name: /place an order/i})).toBeDisabled()
    })
})



