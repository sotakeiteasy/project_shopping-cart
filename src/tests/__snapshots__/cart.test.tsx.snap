// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`render cart > snapshot cart 1`] = `
<div>
  <main>
    <div
      class="_cart_3365fd"
    >
      <div
        class="_products_3365fd"
      >
        <p>
          Cart is empty
        </p>
      </div>
      <div
        class="_checkout_3365fd"
      >
        <h2>
          Placing an order
        </h2>
        <div
          class="_total_3365fd"
        >
          <span>
            Price of ads:
          </span>
          <span>
            0
             $
          </span>
        </div>
        <div
          class="_total_3365fd"
        >
          <span>
            Delivery: 
          </span>
          <span>
            5$
          </span>
        </div>
        <div
          class="_total_3365fd"
        >
          <span>
            Total: 
          </span>
          <span>
            5
            $
          </span>
        </div>
        <button
          class="_checkoutButton_3365fd"
          disabled=""
        >
          Place an order
        </button>
      </div>
    </div>
  </main>
</div>
`;
